packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -18,22 +18,53 @@ import {
   EdmTypeShared,
   isEdmType
 } from '../common';
-import { PropertyMetadata } from '../v2';
+import { EdmToPrimitive, EdmType, PropertyMetadata } from '../v2';
 
 const logger = createLogger({
   package: 'core',
   messageContext: 'entity-deserializer'
 });
 
-// eslint-disable-next-line valid-jsdoc
 /**
- * @experimental This is experimental and is subject to change. Use with caution.
+ * Interface representing the return type of the builder function [[entityDeserializer]]
  */
-export function entityDeserializer(
-  edmToTs,
-  extractODataETag,
-  extractDataFromOneToManyLink: (arg) => any[] | undefined
-) {
+export interface EntityDeserializer<EntityT extends EntityBase = any> {
+  // TODO: extractCustomFields should not be exported here. This was probably done only for testing
```

Comment:
Should we create a BLI for this and remove the comment to get release ready?

Reply:
I will try to remove this. If we export it would be a breaking change.